driver/where_driver.hpp

```diff
@@ -0,0 +1,379 @@
+#include <sys/types.h>
+#include <vector>
+
+#include <../test/verify.hpp>
+
+#include <miopen/errors.hpp>
+#include <miopen/miopen.h>
+
+template <typename Tgpu, typename Tcheck>
+int mloWhereBackwardNoBroadcastRunHost(miopenTensorDescriptor_t outputGradDesc,
+                                       const Tgpu* outputGrad,
+                                       const uint8_t* condition,
+                                       Tcheck* inputGrad,
+                                       Tcheck* otherGrad)
+{
+    auto output_grad_numel = miopen::deref(outputGradDesc).GetElementSize();
+
+    if(inputGrad)
+    {
+        for(size_t i = 0; i < output_grad_numel; i++)
+        {
+            uint8_t cond = (condition[i] > 0) ? 1 : 0;
+            inputGrad[i] = outputGrad[i] * cond;
+        }
+    }
+
+    if(otherGrad)
+    {
+        for(size_t o = 0; o < output_grad_numel; o++)
+        {
+            uint8_t cond = (condition[o] > 0) ? 1 : 0;
+            otherGrad[o] = outputGrad[o] * (1 - cond);
+        }
+    }
+
+    return 0;
+}
+
+inline bool isDefined(const std::vector<int>& len)
+{
+    return std::all_of(len.begin(), len.end(), [](int i) { return i > 0; });
+}
+
+template <typename Tgpu, typename Tref>
+class WhereDriver : public Driver
+{
+public:
+    WhereDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&condTensor);
+        miopenCreateTensorDescriptor(&inputTensorGrad);
+        miopenCreateTensorDescriptor(&otherTensorGrad);
+        miopenCreateTensorDescriptor(&outputTensorGrad);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU(); // Verify implements it
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU(); // Verify implements it
+
+    Tref GetTolerance();
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~WhereDriver() override
+    {
+        miopenDestroyTensorDescriptor(condTensor);
+        miopenDestroyTensorDescriptor(inputTensorGrad);
+        miopenDestroyTensorDescriptor(otherTensorGrad);
+        miopenDestroyTensorDescriptor(outputTensorGrad);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+    bool isInputGradRequired;
+    bool isOtherGradRequired;
+
+    // Backwards
+    miopenTensorDescriptor_t condTensor       = nullptr;
+    miopenTensorDescriptor_t outputTensorGrad = nullptr;
+    miopenTensorDescriptor_t inputTensorGrad  = nullptr;
+    miopenTensorDescriptor_t otherTensorGrad  = nullptr;
+
+    std::unique_ptr<GPUMem> cond_dev      = nullptr;
+    std::unique_ptr<GPUMem> outGrad_dev   = nullptr;
+    std::unique_ptr<GPUMem> inGrad_dev    = nullptr;
+    std::unique_ptr<GPUMem> otherGrad_dev = nullptr;
+
+    std::vector<uint8_t> cond;
```

Comment:
what's the reason of using type `uint8_t` for `cond` tensor instead of `bool`?

Reply:
cause we can't get pointer to elements of `bool` vector (to pass to kernel fucntion)
https://stackoverflow.com/questions/46115669/why-does-stdvectorbool-have-no-data